Trappist/src/Promact.Trappist.Web/wwwroot/app/tests/test-settings/test-settings.component.ts

```diff
@@ -1,65 +1,127 @@
-﻿import { Component, OnInit, ViewChild } from "@angular/core";
-import { MdDialog } from '@angular/material';
-import { Test } from "../tests.model";
-import { TestSettingService } from "../testsetting.service";
+﻿import { Component, OnInit, ViewChild } from '@angular/core';
+import { MdDialog, MdSnackBar } from '@angular/material';
+import { Test } from '../tests.model';
+import { TestService } from '../tests.service';
 import { ActivatedRoute } from '@angular/router';
-
+import { TestLaunchDialogComponent } from './test-launch-dialog.component';
 
 @Component({
     moduleId: module.id,
-    selector: "test-settings",
-    templateUrl: "test-settings.html"
+    selector: 'test-settings',
+    templateUrl: 'test-settings.html'
 })
 
-export class TestSettingsComponent{
-    testsettings: Test = new Test();
+export class TestSettingsComponent implements OnInit {
+    testsettings: Test;
     testId: number;
+    validEndDate: boolean;
+    endDate: string;
+    validTime: boolean;
+    validStartDate: boolean;
+    currentDate: Date;
     editName: string;
+    testNameUpdationMessage: string;
+    settingsUpdationMessage: string;
 
-    // Open Launch Test Dialog
-    constructor(public dialog: MdDialog, private testSettingService: TestSettingService, private route: ActivatedRoute) { }
+    /**
+     * Open Launch Test Dialog
+     * @param dialog is responsible for opening the Dialog box
+     * @param testSettingService is used to get the Url from the testsettings.service file
+     * @param route is used to take the value of Id from the active route
+     */
+    constructor(public dialog: MdDialog, private testService: TestService, private route: ActivatedRoute, private snackbarRef: MdSnackBar) {
+        this.testsettings = new Test();
+        this.validEndDate = false;
+        this.validTime = false;
+        this.validStartDate = false;
+        this.currentDate = new Date();
+        this.testNameUpdationMessage = "Test Name has been successfully updated";
+        this.settingsUpdationMessage = "Data has been updated successfully";
```

Comment:
which data has been updated successfully? Please give proper and user-friendly messages.